File: tests/fixtures/sample_repo/scala_pkg/Calculator.scala
package sample

import sample.models.Operation
import sample.models.CalculationRecord

/** Calculator for basic arithmetic. */
class Calculator extends BaseCalc with Computable {
  def add(x: Double, y: Double): Double = x + y

  def subtract(x: Double, y: Double): Double = x - y

  private def helper(): Unit = {}

  def compute(): Double = 0.0
}

trait Computable {
  def compute(): Double
}

File: tests/fixtures/sample_repo/php_pkg/Calculator.php
<?php
namespace Sample;

use Sample\Models\Operation;
use Sample\Models\CalculationRecord;

/** Calculator for basic arithmetic. */
class Calculator extends BaseCalc implements Computable
{
    public function add(float $x, float $y): float
    {
        return $x + $y;
    }

    public function subtract(float $x, float $y): float
    {
        return $x - $y;
    }

    private function helper(): void {}
}
